src/net/tls.cc

```diff
@@ -1355,27 +1386,38 @@ class session : public enable_lw_shared_from_this<session> {
     }
     ssize_t vec_push(const giovec_t * iov, int iovcnt) {
         if (!_output_pending.available()) {
+            tlslog.debug("vec_push: output pending unavailable");
             gnutls_transport_set_errno(*this, EAGAIN);
             return -1;
         }
-        if (_output_pending.failed()) {
-            // copy exception for on_internal_error
-            auto ex = _output_pending.get_exception();
-            _output_pending = make_exception_future<>(ex);
-            on_internal_error(seastar_logger, format("tls: session: vec_push: unhandled pending output error: {}", ex));
-        }
         try {
-            scattered_message<char> msg;
-            for (int i = 0; i < iovcnt; ++i) {
-                msg.append(std::string_view(reinterpret_cast<const char *>(iov[i].iov_base), iov[i].iov_len));
+            ssize_t n;
+            bool pending_error = _output_pending.failed();
+
+            if (!pending_error) {
+                scattered_message<char> msg;
+                for (int i = 0; i < iovcnt; ++i) {
+                    msg.append(std::string_view(reinterpret_cast<const char *>(iov[i].iov_base), iov[i].iov_len));
+                }
+                n = msg.size();
+                _output_pending = _out.put(std::move(msg).release());
             }
-            auto n = msg.size();
-            _output_pending = _out.put(std::move(msg).release());
             if (_output_pending.failed()) {
+                auto ex = _output_pending.get_exception();
                 // exception is copied back into _output_pending
                 // by the catch handlers below
-                std::rethrow_exception(_output_pending.get_exception());
+                auto level = pending_error ? log_level::warn : log_level::debug;
+                tlslog.log(level, "tls: vec_push: {}output failed: {}", pending_error ? "pending " : "", ex);
+                std::rethrow_exception(ex);
+            }
+#ifdef SEASTAR_DEBUG
+            if (_inject_vec_push_async_error && !_output_pending.available()) {
+                tlslog.warn("vec_push: injecting async output error");
+                _output_pending = _output_pending.then([] {
+                    return make_exception_future<>(std::runtime_error("injected vec_push error"));
+                });
             }
```

Comment:
Again, can accomplish this by just writing the test to run on a pipe and inject the errors there.